### PR TITLE
Correctly support Chiseled Bookshelf in Transfer API

### PR DESCRIPTION
**Pull request by Technici4n**

### CODE REVIEW

File: fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/gametests/VanillaStorageTests.java [hunk 1]
```diff
@@ -123,6 +128,85 @@ public void testChestComparator(TestContext context) {
...
+		if (bookshelf.method_47887() != 1) throw new GameTestException("Last modified slot should be 1 after committing transaction");
+
+		// Let's also check the state properties. Only slot 2 should be occupied.
+		BlockState state = bookshelf.getCachedState();
+
+		if (state.get(Properties.SLOT_0_OCCUPIED)) throw new GameTestException("Slot 0 should not be occupied");
+		if (state.get(Properties.SLOT_1_OCCUPIED)) throw new GameTestException("Slot 1 should not be occupied");
+		if (!state.get(Properties.SLOT_2_OCCUPIED)) throw new GameTestException("Slot 2 should be occupied");
+		if (state.get(Properties.SLOT_3_OCCUPIED)) throw new GameTestException("Slot 3 should be occupied");
+		if (state.get(Properties.SLOT_4_OCCUPIED)) throw new GameTestException("Slot 4 should be occupied");
```
Comment (Shnupbups) [inline fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/gametests/VanillaStorageTests.java]: Exception text for tests on slots 3 and 4 are inverted

Comment (Technici4n) [reply fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/gametests/VanillaStorageTests.java]: Good catch, thank you! :smile:

**Comment by modmuss50** [comment]
Note for future self: Rebase and merge this

File: fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/gametests/VanillaStorageTests.java [hunk 2]
```diff
@@ -123,6 +128,85 @@ public void testChestComparator(TestContext context) {
...
+		InventoryStorage storage = InventoryStorage.of(bookshelf, null);
+
+		// First, check that we can correctly undo insert operations, because vanilla's setStack doesn't permit it without our patches.
+		try (Transaction transaction = Transaction.openOuter()) {
+			if (storage.insert(book, 2, transaction) != 2) throw new GameTestException("Should have inserted 2 books");
+
+			if (bookshelf.getStack(0).getCount() != 1) throw new GameTestException("Bookshelf stack 0 should have size 1");
+			if (!book.matches(bookshelf.getStack(0))) throw new GameTestException("Bookshelf stack 0 should be a book");
+			if (bookshelf.getStack(1).getCount() != 1) throw new GameTestException("Bookshelf stack 1 should have size 1");
+			if (!book.matches(bookshelf.getStack(1))) throw new GameTestException("Bookshelf stack 1 should be a book and quill");
```
Comment (Shnupbups) [inline fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/gametests/VanillaStorageTests.java]: Doesn't seem like a Book and Quill is ever inserted, nor does this statement check for one, so the exception text is wrong.

Comment (Technici4n) [reply fabric-transfer-api-v1/src/testmod/java/net/fabricmc/fabric/test/transfer/gametests/VanillaStorageTests.java]: You're right. I used to check that call with a book and quill, but I removed it. Thanks again.

**Review by modmuss50 — APPROVED** [review]

**Pull request merged**